tests/components/homematicip_cloud/test_config_flow.py

```diff
@@ -21,6 +21,8 @@
     with patch.object(hap, 'get_auth', return_value=mock_coro()), \
```

Comment:
I think the problem is that we're patching this local instance `hap`. That isn't the same instance that is used in the code under test. We should patch the methods of the code where it's used.

Reply:
But this is patching the used hap object!? At least seems to work.

Reply:
We're just patching the `hap` object we created in this test on the line above. It's not the same object that will be created and used when we set up the component. We have to patch the code under test.